src/electionguard/encrypt.cpp

```diff
@@ -302,22 +304,93 @@ namespace electionguard
+            for (const auto &selection : selections) {
+                if (selection.get().getVote() == 1) {
+                    errorData.push_back(selection.get().getObjectId());
+                }
+            }
+            extendedData["errorData"] = errorData;
+        }
+
+        json writeins;
+        auto candidates = internalManifest.getCandidates();
+        std::vector<std::reference_wrapper<SelectionDescription>> ballotSelections;
+
+        // find the contest in the manifest
+        for (const auto &manifestContest : internalManifest.getContests()) {
+            if (contest.getObjectId() == manifestContest.get().getObjectId()) {
+                ballotSelections = manifestContest.get().getSelections();
+            }
+        }
+
+        // run through the selections in this contest and see if any of them are writeins
+        // the number of selections should be short, the number of ballot selections
+        // and candidates will be longer but shouldn't be too long
+        for (const auto &selection : selections) {
+            if (selection.get().getVote() == 1) {
+                for (const auto ballotSelection : ballotSelections) {
+                    if (selection.get().getObjectId() == ballotSelection.get().getObjectId()) {
+                        for (const auto &candidate : candidates) {
+                            // check if the candidate is the correct one and if it is the writein option
+                            if (ballotSelection.get().getCandidateId() ==
+                                candidate.get().getObjectId()) {
+                                if (candidate.get().isWriteIn()) {
+                                    writeins[selection.get().getObjectId()] =
+                                      selection.get().getWriteIn();
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        if (writeins.dump() != string("null")) {
+            extendedData["writeins"] = writeins;
+        }
+
+        string extendedData_string("");
+        if (extendedData.dump() != string("null")) {
+            extendedData_string = extendedData.dump();      
+        }
+
+        return extendedData_string;
     }
 
     unique_ptr<CiphertextBallotContest>
     encryptContest(const PlaintextBallotContest &contest,
+                   const InternalManifest &internalManifest,
                    const ContestDescriptionWithPlaceholders &description,
                    const ElementModP &elgamalPublicKey, const ElementModQ &cryptoExtendedBaseHash,
                    const ElementModQ &nonceSeed, bool shouldVerifyProofs /* = true */)
 
     {
         // Validate Input
-        if (!contest.isValid(description.getObjectId(), description.getSelections().size(),
-                             description.getNumberElected(), description.getVotesAllowed())) {
+        bool supportOvervotes = true;
```

Comment:
It is currently on by default because of that true. This code was put in because in the future we may have configuration to turn it off, as Python does. Currently it is always on in the C++ repo.